### PR TITLE
Refactor services into builder functions and an api folder

### PR DESCRIPTION
**Pull request by krish-andres**
Refactor services into builder functions and an api folder

### CODE REVIEW

**Comment by krish-andres** [comment]
Note: We haven't edited the auth_service.js file yet

File: assets/js/services/auth_service.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 app.factory("AuthService", ["$http", "$location", function($http, $location) {
+
```
Comment (tesk9) [inline assets/js/services/auth_service.js]: these spaces seem inconsistent with what we've got in the controllers and elsewhere.

File: assets/js/services/library_builder.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    return Library.get({id: id});
+  };
+
+  var addModule = function(name, libraryId, cb) {
+    var module = new Module({name: name, library: libraryId});
+    module.$save(cb); 
+  };
+
+  var removeModule = function(moduleId) {
+    var module = Module.get({id: moduleId});
```
Comment (sdevani) [inline assets/js/services/library_builder.js]: The get request will be asynchronous, so this should not work. Also why not just make a call to remove by giving it an id? Module.remove({id: moduleId})?

File: assets/js/services/lesson_builder.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  var editLesson = function(body, lesssonId) {
+    var lesson = Lesson.get({id: lessonId}, function(data) {
+      data.body = body;
+      data.$update();
+    });
+  };
+
+  return {
+    editLesson: editLesson
+  }
```
Comment (tesk9) [inline assets/js/services/lesson_builder.js]: ;

File: assets/js/services/library_builder.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      data.$update();
+    });
+  };
+
+  return {
+    library: library, 
+    addModule: addModule, 
+    removeModule: removeModule,
+    editModuleName: editModuleName
+  }
```
Comment (tesk9) [inline assets/js/services/library_builder.js]: ;

File: assets/js/services/lesson_builder.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+app.factory("LessonBuilder", ["Lesson", function(Lesson) {
+
+  var editLesson = function(body, lesssonId) {
+    var lesson = Lesson.get({id: lessonId}, function(data) {
```
Comment (sdevani) [inline assets/js/services/lesson_builder.js]: You should not do a get request before updating

File: assets/js/services/team_builder.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      data.$update(cb);
+    });
+  };
+
+  return: {
+    createTeam: createTeam,
+    team: team, 
+    addUser: addUser, 
+    addAdmin: addAdmin
+  }
```
Comment (tesk9) [inline assets/js/services/team_builder.js]: ;

File: assets/js/services/team_builder.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+app.factory("TeamBuilder", ["Team", function(Team) {
+
```
Comment (tesk9) [inline assets/js/services/team_builder.js]: whitespace

File: assets/js/services/module_builder.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    var lesson = new Lesson({title: title, module: moduleId});
+    lesson.$save(cb);
+  };
+
+  var removeLesson = function(lessonId) {
+    var lesson = Lesson.get({id: lessonId});
+    lesson.$remove();
+  };
+
+
```
Comment (tesk9) [inline assets/js/services/module_builder.js]: whitespace

File: assets/js/services/module_builder.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+app.factory("ModuleBuilder", ["Module", "Lesson", function(Module, Lesson) {
+
```
Comment (tesk9) [inline assets/js/services/module_builder.js]: whitespace

File: assets/js/services/library_builder.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
+app.factory("LibraryBuilder", ["Library", "Module", function(Library, Module) {
+
```
Comment (tesk9) [inline assets/js/services/library_builder.js]: whitespace

File: assets/js/services/module_builder.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    return Module.get({id: id});
+  };
+
+  var addLesson = function(title, moduleId, cb) {
+    var lesson = new Lesson({title: title, module: moduleId});
+    lesson.$save(cb);
+  };
+
+  var removeLesson = function(lessonId) {
+    var lesson = Lesson.get({id: lessonId});
```
Comment (sdevani) [inline assets/js/services/module_builder.js]: Again, you shouldn't call get in order to remove

File: assets/js/services/lesson_builder.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+app.factory("LessonBuilder", ["Lesson", function(Lesson) {
+
```
Comment (tesk9) [inline assets/js/services/lesson_builder.js]: whitespace

File: assets/js/services/team_builder.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+  var createTeam = function(name, cb) {
+    var team = new Team({name: name});
```
Comment (sdevani) [inline assets/js/services/team_builder.js]: Will the backend be responsible for creating the library?

Comment (krish-andres) [inline assets/js/services/team_builder.js]: I believe so

File: assets/js/services/lesson_builder.js (hunk 5 above)
Comment (krish-andres) [inline assets/js/services/lesson_builder.js]: How else should we update? This is the way that the learn-app shows how to update...

File: assets/js/services/team_builder.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  var addUser = function(teamId, user, cb) {
+    var team = Team.get({id: teamId}, function(data) {
+      data.users.push(user);
+      data.$update(cb);
+    });
+  };
+
+  var addAdmin = function(teamId, user, cb) {
```
Comment (sdevani) [inline assets/js/services/team_builder.js]: If the team has already been loaded beforehand, can we have a function such as:
`
var addAdmin = function(team, user, cb) {
team.admins.push(user);
team.users.push(user);
team.$update(cb);
};
`

Or should we actually have a backend route for this:

`POST '/team/5/add-admin'`
we can send the admin id as data in the post request

File: assets/js/services/lesson_builder.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+  var lesson = function(id) {
```
Comment (sdevani) [inline assets/js/services/lesson_builder.js]: Remove this function.

File: assets/js/services/lesson_builder.js [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+    return Lesson.get({id: id});
+  };
+
+  var editLesson = function(body, lesson) {
```
Comment (sdevani) [inline assets/js/services/lesson_builder.js]: intuitive convention is to switch lesson and body

File: assets/js/services/library_builder.js [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+  var library = function(id) {
+    return Library.get({id: id});
+  };
+
+  var addModule = function(name, libraryId, cb) {
```
Comment (sdevani) [inline assets/js/services/library_builder.js]: This should update the stateTracker if necessary.

**Pull request closed without merging**